packages/eslint-plugin/src/rules/ban-ts-comment.ts

```diff
@@ -38,43 +38,45 @@ export default util.createRule<[Options], MessageIds>({
       tsDirectiveCommentDescriptionNotMatchPattern:
         'The description for the "@ts-{{directive}}" directive must match the {{format}} format.',
     },
-    schema: [
-      {
-        definitions: {
-          directiveConfigSchema: {
-            oneOf: [
-              {
-                type: 'boolean',
-                default: true,
+    schema: {
+      definitions: {
+        directiveConfigSchema: {
+          oneOf: [
+            {
+              type: 'boolean',
+              default: true,
+            },
+            {
+              enum: ['allow-with-description'],
+            },
+            {
+              type: 'object',
+              properties: {
+                descriptionFormat: { type: 'string' },
               },
-              {
-                enum: ['allow-with-description'],
-              },
-              {
-                type: 'object',
-                properties: {
-                  descriptionFormat: { type: 'string' },
-                },
-              },
-            ],
-          },
+            },
+          ],
         },
-        type: 'object',
-        properties: {
-          'ts-expect-error': {
-            $ref: '#/definitions/directiveConfigSchema',
-          },
-          'ts-ignore': { $ref: '#/definitions/directiveConfigSchema' },
-          'ts-nocheck': { $ref: '#/definitions/directiveConfigSchema' },
-          'ts-check': { $ref: '#/definitions/directiveConfigSchema' },
-          minimumDescriptionLength: {
-            type: 'number',
-            default: defaultMinimumDescriptionLength,
+      },
+      prefixItems: [
+        {
+          properties: {
+            'ts-expect-error': {
+              $ref: '#/definitions/directiveConfigSchema',
+            },
+            'ts-ignore': { $ref: '#/definitions/directiveConfigSchema' },
+            'ts-nocheck': { $ref: '#/definitions/directiveConfigSchema' },
+            'ts-check': { $ref: '#/definitions/directiveConfigSchema' },
```

Comment:
_Sigh_. Working on it.

<img width="871" alt="Screenshot of ban-ts-comments with type Options = unknown" src="https://user-images.githubusercontent.com/3335181/186174608-ac6882e3-bffb-47f0-ba86-14354d21276e.png">